### PR TITLE
Updating the adapter to use the new rewarded API.

### PR DESCRIPTION
**Pull request by maninomula**

### CODE REVIEW

File: AdMob/MPGoogleAdMobRewardedVideoCustomEvent.m [hunk 1]
```diff
@@ -114,56 +124,36 @@ - (void)handleAdPlayedForCustomEventNetwork {
...
+  [self.delegate rewardedVideoWillDisappearForCustomEvent:self];
 
-    MPLogAdEvent([MPLogEvent adDidDisappearForAdapter:NSStringFromClass(self.class)], [self getAdNetworkId]);
-    [self.delegate rewardedVideoDidDisappearForCustomEvent:self];
-}
-
-- (void)rewardBasedVideoAdWillLeaveApplication:(GADRewardBasedVideoAd *)rewardBasedVideoAd {
-    // Recording a click because the rewardBasedVideoAdWillLeaveApplication: is invoked when a click
-    // on the reward-based video ad happens.
-    [self.delegate trackClick];
```
Comment (ericleich) [inline AdMob/MPGoogleAdMobRewardedVideoCustomEvent.m]: MoPub is going to be sad about this. Unfortunately we no longer have a good click event anymore.

File: AdMob/CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,8 @@
 ## Changelog
+  * 7.42.2.0
+    * Updated the adapter to use Google's new rewarded API.
+    * This version of the adapters has been certified with AdMob 7.42.2.
```
Comment (chauduyphanvu1) [inline AdMob/CHANGELOG.md]: Please add an entry pertaining to the new targeting code.

File: AdMob/MPGoogleAdMobRewardedVideoCustomEvent.m [hunk 3]
```diff
@@ -66,27 +75,35 @@ - (void)requestRewardedVideoWithCustomEventInfo:(NSDictionary *)info {
...
+        MPLogAdEvent([MPLogEvent adLoadSuccessForAdapter:NSStringFromClass(self.class)], [self getAdNetworkId]);
+        [self.delegate rewardedVideoDidLoadAdForCustomEvent:self];
+      }
+    }];
     MPLogAdEvent([MPLogEvent adLoadAttemptForAdapter:NSStringFromClass(self.class) dspCreativeId:nil dspName:nil], [self getAdNetworkId]);
 }
 
 - (BOOL)hasAdAvailable {
-    return [GADRewardBasedVideoAd sharedInstance].isReady;
+    return self.rewardedAd.isReady;
```
Comment (chauduyphanvu1) [inline AdMob/MPGoogleAdMobRewardedVideoCustomEvent.m]: Can `self.rewardedAd` be `nil` here? If so, it's recommended to guard against that.

File: AdMob/MPGoogleAdMobRewardedVideoCustomEvent.m [hunk 4]
```diff
@@ -114,56 +131,36 @@ - (void)handleAdPlayedForCustomEventNetwork {
...
     MPLogAdEvent([MPLogEvent adShowSuccessForAdapter:NSStringFromClass(self.class)], [self getAdNetworkId]);
     MPLogAdEvent([MPLogEvent adDidAppearForAdapter:NSStringFromClass(self.class)], [self getAdNetworkId]);
 }
 
-- (void)rewardBasedVideoAdDidStartPlaying:(GADRewardBasedVideoAd *)rewardBasedVideoAd {
-    // MoPub rewarded video custom event doesn't have a callback when the video starts playing.
-    MPLogInfo(@"Google AdMob reward-based video ad started playing.");
+- (void)rewardedAd:(GADRewardedAd *)rewardedAd didFailToPresentWithError:(NSError *)error {
+    [self.delegate rewardedVideoDidFailToPlayForCustomEvent:self error:error];
+    MPLogAdEvent([MPLogEvent adShowFailedForAdapter:NSStringFromClass(self.class) error:error], [self getAdNetworkId]);
```
Comment (chauduyphanvu1) [inline AdMob/MPGoogleAdMobRewardedVideoCustomEvent.m]: Let's log first before calling the delegate just to be safe (here and wherever applicable).

**Pull request merged**